src/services/dumpers/agent-nodejs.ts

```diff
@@ -82,29 +83,66 @@ export default class AgentNodeJs extends AbstractDumper {
       }
     }
 
-    const scripts: { [name: string]: string } = {
+    let scripts: { [name: string]: string } = {
       start: 'nodemon ./index.js',
       'start:agent': 'node ./index.js',
     };
     const devDependencies: { [name: string]: string } = {
       nodemon: '^2.0.12',
     };
+    const nodemonConfig = {
+      ignore: ['./forestadmin-schema.json'],
+    };
+
+    if (language === languages.Typescript) {
+      scripts = {
+        build: 'tsc',
+        start: 'nodemon ./index.ts',
+        'start:agent': 'node ./dist/index.js',
+      };
+      devDependencies.typescript = '^4.9.4';
+      devDependencies['ts-node'] = '^10.9.1';
+      nodemonConfig.ignore.push('./typings.ts');
+    }
 
     const pkg = {
       name: this.toValidPackageName(appName),
       version: '0.0.1',
       private: true,
       scripts,
-      nodemonConfig: {
-        ignore: ['./forestadmin-schema.json'],
-      },
+      nodemonConfig,
       dependencies,
       devDependencies,
     };
 
     this.writeFile('package.json', `${JSON.stringify(pkg, null, 2)}\n`);
   }
 
+  writeTsConfigJson() {
+    this.writeFile(
+      'tsconfig.json',
+      `${JSON.stringify(
+        {
+          compilerOptions: {
+            experimentalDecorators: true,
+            target: 'ES2020',
+            module: 'CommonJS',
+            moduleResolution: 'Node',
+            esModuleInterop: true,
+            declaration: true,
+            declarationMap: true,
+            inlineSourceMap: true,
+            noImplicitOverride: true,
+            stripInternal: true,
+            outDir: 'dist',
+          },
+        },
+        null,
+        2,
+      )}\n`,
+    );
+  }
+
```

Comment:
Watch out, you did not change the datasource import in `index.ts`, translate it to typescript format 👌